addons/sale_project/models/project.py

```diff
@@ -772,6 +772,11 @@ class ProjectTask(models.Model):
     # Project sharing  fields
     display_sale_order_button = fields.Boolean(string='Display Sales Order', compute='_compute_display_sale_order_button')
 
+    def write(self, vals):
+        if len(self) > 1 and 'sale_line_id' in vals and not (len(self.partner_id.ids) == 1 and all(task.partner_id for task in self)):
+            raise ValidationError(_("To update the sales order item in batch, all selected tasks must be associated with the same customer."))
+        return super().write(vals)
```

Comment:
Can we not instead prevent this situation from happening in the first place? Or is that not worth a JS customization?

Reply:
I don't think I would be feasible to add js custom for this because preventing user to not allow multi editable for just for on field and for specific condition like having a different partner or not.
Don't you think? :thinking:

Reply:
We already do that for the stage for tasks in different projects, see `odoo/addons/project/static/src/views/project_task_list/project_task_list_renderer.js`

Reply:
Hello @MissingNoShiny
Indeed its my bad i didn't see it
changes done :slightly_smiling_face: